### ISSUE TITLE
Release midr 0.5.0

### ISSUE THREAD

**Issue by ryo-asashi** [ryo-asashi/midr#1]
First release:

* [x] `usethis::use_cran_comments()`
* [x] Update (aspirational) install instructions in README
* [x] Proofread `Title:` and `Description:`
* [x] Check that all exported functions have `@return` and `@examples`
* [x] Check that `Authors@R:` includes a copyright holder (role 'cph')
* [x] Check [licensing of included files](https://r-pkgs.org/license.html#sec-code-you-bundle)
* [x] Review <https://github.com/DavisVaughan/extrachecks>

Prepare for release:

* [x] `git pull`
* [x] `urlchecker::url_check()`
* [x] `devtools::build_readme()`
* [x] `devtools::check(remote = TRUE, manual = TRUE)`
* [x] `devtools::check_win_devel()`
* [x] `git push`

Submit to CRAN:

* [x] `usethis::use_version('minor')`: -> incremented to 0.5.0
* [x] `devtools::submit_cran()`
* [x] Approve email

Wait for CRAN...

* [x] Accepted :tada:
* [x] `usethis::use_github_release()`
* [ ] `usethis::use_dev_version(push = TRUE)`

**Comment by ryo-asashi** [comment]
- [x]  Improve help documents.
- [x]  Write an `.Rmd` vignette for the package. --> We have submitted one for JARIP Bulletin.
- [x]  Submit `midr` to CRAN by January 12, 2025.
- [x]  Fix `factor.frame()` / `numeric.frame()` so that the resulting dataframes can be used with dplyr functions.
- [x]  Clean up unnecessary scripts and the midr.pdf.
- [x] Fix the example of `weighted()` that does not run successfully. <- this issue was not caused by the example code and/or the GitHub Action environment, but by the functions `shuffled()`, `augmented()` and `latticized()` themselves.

**Comment by ryo-asashi** [comment]
- [x] Fix the bug in `model.reframe`; data <- data[, -which(colnames(data) == yvar)**, drop = FALSE**]
- [x] Fix the example code of 'interpret()': remove "+" from "+ points()".
- [x] Define `print.encoder()` to improve useability of fitted MID models.

**Comment by ryo-asashi** [comment]
- [x] Add methods of the `get.yhat()` function for some important model classes incl. `xgboost` and `lightgbm`. This requires a careful modification of the `interpret()` function that needs to take a matrix as a valid input.
- [x] Let 'mid' object have information about the model class.

**Comment by ryo-asashi** [comment]
- [x] Fix the bug of `ggmid()` and `plot()` for the cases the catchall level is used and the argument `theme` is set at `data` or `compound`.

**Comment by ryo-asashi** [comment]
Please add \value to .Rd files regarding exported methods and explain
the functions results in the documentation. Please write about the
structure of the output (class) and also what the output means. (If a
function does not return a value, please document that too, e.g.
\value{No return value, called for side effects} or similar)
For more details:
<https://contributor.r-project.org/cran-cookbook/docs_issues.html#missing-value-tags-in-.rd-files>

->  Missing Rd-tags:
      scale_color_theme.Rd: \value

Please always make sure to reset to user's options(), working directory
or par() after you changed it in examples and vignettes and demos.
e.g.:
oldpar <- par(mfrow = c(1,2))
...
par(oldpar)

-> man/color.theme.Rd

For more details:
<https://contributor.r-project.org/cran-cookbook/code_issues.html#change-of-options-graphical-parameters-and-working-directory>
